conans/client/proxy.py

```diff
@@ -395,13 +395,22 @@ def _retrieve_remote_package(self, package_ref, package_folder, output, remote=N
             output.warn('Binary for %s not in remote: %s' % (package_id, str(e)))
             return False
 
-    def authenticate(self, name, password):
-        if not name:  # List all users, from all remotes
-            remotes = self._registry.remotes
-            if not remotes:
-                self._out.error("No remotes defined")
-            for remote in remotes:
-                self._remote_manager.authenticate(remote, None, None)
-            return
-        remote, _ = self._get_remote()
-        return self._remote_manager.authenticate(remote, name, password)
+    def authenticate(self, name, password, all_remotes=False):
+        """
+        Manage user auth against remote.
+        Also displays a list of authenticated users against remote(s) if user name is evaluated to False.
+
+        :param name: user name string
+        :param password: password string
+        :param all_remotes: True/False to use all available remotes to display a list of authenticated users if
+                            user name is evaluated to False.
+        """
+        current_remote, _ = self._get_remote()
+
+        if name:
+            return self._remote_manager.authenticate(current_remote, name, password)
+
+        # List all users from required remotes
+        remotes = self._registry.remotes if all_remotes else [current_remote]
```

Comment:
This code is evidencing a poor design in the current codebase, because it is doing 2 different things under the same name, the second on (listing users) not corresponding to the method name. It would have been good to improve this design.